grace/models/train.py

```diff
@@ -29,33 +48,62 @@ def train_model(
         # weight_decay=5e-4),
     )
 
-    criterion = torch.nn.CrossEntropyLoss()
+    node_criterion = torch.nn.CrossEntropyLoss(
+        ignore_index=Annotation.UNKNOWN, size_average=True
+    )
```

Comment:
I'm not sure what happens if the whole subgraph is unknown?

Reply:
@quantumjot In that case the node loss reduces to NaN... same behaviour with the `edge_criterion` custom loss; would it be worth adding `loss_node = torch.nan_to_num(loss_node)` etc?

Reply:
Ah, like some if statement, e.g.

`if torch.all(torch.where(data.edge_label == Annotation.UNKNOWN, True, False)):
    continue`

?

Reply:
```
edge_label = [
          edge[GraphAttrs.EDGE_GROUND_TRUTH]
          for _, _, edge in sub_graph.edges(data=True)
      ]

if all([e == Annotation.UNKNOWN for e in edge_label]):
   continue
```
Would this work?

Reply:
@quantumjot in `datasets.py`, that is

Reply:
Yeah - I think that works. It may also need to have this if all the nodes are empty.

Reply:
No, that the nodes (i.e. `GraphAttrs.NODE_GROUND_TRUTH]`) are also all `UNKNOWN`.

Reply:
Cool, will add that as well